src/main/java/org/skife/jdbi/v2/unstable/BindIn.java

```diff
@@ -85,6 +86,42 @@ public void apply(SQLStatement q) throws SQLException
                 }
             };
         }
+
+		private static Collection<?> tryToCast(Object arg) {
+			if (arg instanceof Collection) {
+				return (Collection<?>) arg;
+			}
+			if (arg instanceof Object[]) {
+				return Arrays.asList((Object[]) arg);
+			}
+			// necessary because of how java handles primitives...
+			if (arg instanceof boolean[]) {
+				return Arrays.asList((boolean[]) arg);
+			}
+			if (arg instanceof byte[]) {
+				return Arrays.asList((byte[]) arg);
```

Comment:
Are you sure this construct actually works?  I believe this is actually creating a `List<byte[]>` which is not what you want.